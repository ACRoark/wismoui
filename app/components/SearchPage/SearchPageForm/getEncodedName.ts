const getEncodedName = (name: string): string =>
  encodeURIComponent(name).replace(
    /[-_.!~*'()]/g,
    (character: string): string => `%${character.charCodeAt(0).toString(16).toUpperCase()}`,
  );

export default getEncodedName;
