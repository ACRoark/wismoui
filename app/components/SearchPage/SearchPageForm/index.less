@import '../../../styles/themes/index.less';

@search-page-form-cls-prefix: ~'@{dte-prefix}-search-page-form';

.@{search-page-form-cls-prefix} {
  display: flex;
  flex-direction: column;

  .title {
    font-size: @dte-wismo-font-size-1;
    line-height: 1.875rem;
    text-align: center;
  }

  .recaptcha {
    display: flex;
    justify-content: center;
    transform: scale(1.08);
  }

  .button {
    align-self: center;
    background-color: @dte-blue-light;
    border: none;
    color: @dte-white;
    cursor: pointer;
    font-size: @dte-wismo-font-size-1;
    font-weight: @dte-wismo-font-weight-semi-bold;
    height: 40px;
    margin-top: 30px;
    width: 142px;

    &:hover {
      background-color: @dte-blue;
    }
  }

  .button:disabled {
    background-color: @dte-gray-500;
    cursor: not-allowed;
  }

  input {
    border: 1px solid @dte-gray-400;
    height: 36px;
    outline: none;
    padding-left: 12px;
    padding-right: 12px;
    width: 100%;

    &[name='order-number'] {
      margin-bottom: 30px;
    }

    &:focus {
      border: 1px solid @dte-blue;
    }
  }

  &-input-header {
    margin-bottom: 4px;
  }

  label {
    color: @dte-gray-600;
    font-size: @dte-wismo-font-size-s;
    font-weight: @dte-wismo-font-weight-semi-bold;
    margin-top: 30px;
  }
}
