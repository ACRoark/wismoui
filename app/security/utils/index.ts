import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
