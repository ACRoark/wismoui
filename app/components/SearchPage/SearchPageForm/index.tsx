import React, { FC, ReactElement, RefObject, SyntheticEvent, useEffect, useState } from 'react';
import ReCAPTCHA from 'react-google-recaptcha';
import InputMask from 'react-input-mask';
import { FormattedMessage } from 'react-intl';
import { useLocation } from 'react-router-dom';

import useConfig from 'hooks/useConfig';
import useFlags from 'hooks/useFlags';
import useThunkDispatch from 'hooks/useThunkDispatch';
import encryptionApiClient from 'services/encryptionApiClient';
import { findOrder, resetTracker } from 'store/actions';
import SearchFormError from '../SearchFormError';
import getEncodedName from './getEncodedName';
import messages from './messages';

import './index.less';

interface ISearchPageFormProps {
  error?: string;
}

const SearchPageForm: FC<ISearchPageFormProps> = (props: ISearchPageFormProps): ReactElement => {
  const { error } = props;

  const { bug2263 } = useFlags();

  const dispatch = useThunkDispatch();
  const [isDisabled, setIsDisabled] = useState(true);
  const [name, setName] = useState('');
  const [orderNumber, setOrderNumber] = useState('');
  const { search: queryParameter = '' } = useLocation();
  const recaptchaRef: RefObject<any> = React.createRef(); // tslint:disable-line
  const {
    recaptchaSitekey,
    urls: { decryptTokenApi },
  } = useConfig();
  const [recaptchaToken, setRecaptchaToken] = useState('');

  const handleDisabled = (): void =>
    name.length && orderNumber.length && recaptchaToken ? setIsDisabled(false) : setIsDisabled(true);

  const handleNameChange = (event: SyntheticEvent): void => {
    const element = event.target as HTMLInputElement;

    setName(element.value);
  };

  const handleOrderNumberChange = (event: SyntheticEvent): void => {
    const element = event.target as HTMLInputElement;

    setOrderNumber(element.value);
  };

  const handleSubmit = async (event: SyntheticEvent): Promise<void> => {
    event.preventDefault();

    dispatch(findOrder(orderNumber.replace(/\s/g, ''), getEncodedName(name), recaptchaToken, bug2263));
    setIsDisabled(true);
  };

  const recaptchaOnChange = (value: string): void => {
    setRecaptchaToken(value || '');
    dispatch(resetTracker());
  };

  useEffect((): void => {
    if (queryParameter) {
      const setDecryptedOrderNumber = async (): Promise<void> => {
        const decryptedOrderNumber = await encryptionApiClient.decryptTokenAsync(decryptTokenApi, queryParameter);

        setOrderNumber(decryptedOrderNumber);
      };

      setDecryptedOrderNumber();
    }
  }, []);

  useEffect((): void => {
    handleDisabled();
  }, [recaptchaToken, name, orderNumber]);

  useEffect((): void => {
    if (error && recaptchaRef.current) {
      recaptchaRef.current.reset();
      setRecaptchaToken('');
    }
  }, [error]);

  return (
    <form
      className="dte-wismo-search-page-form"
      id="mimotrackerguestsearch"
      name="mimotrackerguestsearch"
      onSubmit={handleSubmit}
    >
      <h1 className="title">
        <FormattedMessage {...messages.title} />
      </h1>

      <label htmlFor="name">
        <div className="dte-wismo-search-page-form-input-header">
          <FormattedMessage {...messages.name} />
        </div>
        <input name="name" type="text" value={name} onChange={handleNameChange} />
      </label>

      <label htmlFor="order-number">
        <div className="dte-wismo-search-page-form-input-header">
          <FormattedMessage {...messages.orderNumber} />
        </div>
        <InputMask
          mask="** **** ****"
          maskChar={null}
          name="order-number"
          onChange={handleOrderNumberChange}
          value={orderNumber}
        />
      </label>

      <ReCAPTCHA className="recaptcha" ref={recaptchaRef} sitekey={recaptchaSitekey} onChange={recaptchaOnChange} />

      {error && <SearchFormError error={error} />}

      <button
        className="button"
        data-track="mimo"
        data-track-detail="order tracker"
        data-track-action="click"
        data-track-sub-action="guest search"
        type="submit"
        disabled={isDisabled}
      >
        <FormattedMessage {...messages.button} />
      </button>
    </form>
  );
};

export default SearchPageForm;
