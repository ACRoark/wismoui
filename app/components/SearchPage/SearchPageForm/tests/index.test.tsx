import React from 'react';
import { Provider } from 'react-redux';
import routeData from 'react-router';

import { SUPPORTED_LOCALES } from 'locales';

import { createMockStore, createSnapshotWithIntl  } from 'testing/utils';
import SearchPageForm from '..';

const store = createMockStore();

const errors = ['ORDER_INACTIVE', 'ORDER_INVALID', 'ORDER_NOT_FOUND', 'RECAPTCHA_FAILED', 'SERVER_UNAVAILABLE', 'well this was unexpected' ];

const mockLocation = {
  pathname: '/welcome',
  hash: '',
  search: '',
  state: ''
};

beforeEach((): void => {
  jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation);
});

const testSearchPageFormSnapshotWithError = ( error: string, language: string): void => {
  const tree = createSnapshotWithIntl(<Provider store={store}><SearchPageForm error={error} /></Provider>, language);

  expect(tree).toMatchSnapshot();
};

const testSearchPageFormSnapshotWithOutError = (language: string): void => {
  const tree = createSnapshotWithIntl(<Provider store={store}><SearchPageForm /></Provider>, language);

  expect(tree).toMatchSnapshot();
};

describe('<SearchPageForm />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    errors.forEach((error: string): void => {
      it(`should render correctly when error is ${error} and language is ${language}`, (): void => {
        testSearchPageFormSnapshotWithError(error, language);
      });
    });

    it(`should render correctly when language is ${language}`, (): void => {
      testSearchPageFormSnapshotWithOutError( language);
    });
  });
});
