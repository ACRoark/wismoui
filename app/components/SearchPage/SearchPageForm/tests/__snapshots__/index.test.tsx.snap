// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchPageForm /> should render correctly when error is ORDER_INACTIVE and language is en 1`] = `
<form
  className="dte-wismo-search-page-form"
  id="mimotrackerguestsearch"
  name="mimotrackerguestsearch"
  onSubmit={[Function]}
>
  <h1
    className="title"
  >
    To check the status of your order, please enter your information below.
  </h1>
  <label
    htmlFor="name"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Last Name or Business Name
    </div>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label
    htmlFor="order-number"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Order Number
    </div>
    <input
      name="order-number"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onPaste={[Function]}
      value=""
    />
  </label>
  <div
    className="recaptcha"
  />
  <div
    className="dte-wismo-search-form-error"
  >
    <div
      className="message"
    >
      <span
        className="icon"
      >
        !
      </span>
      The information provided is not associated with an active order. Please try again.
    </div>
    <button
      className="close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="18"
        viewBox="0 0 20 20"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
          stroke="currentColor"
          strokeLinecap="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </div>
  <button
    className="button"
    data-track="mimo"
    data-track-action="click"
    data-track-detail="order tracker"
    data-track-sub-action="guest search"
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`<SearchPageForm /> should render correctly when error is ORDER_INACTIVE and language is es 1`] = `
<form
  className="dte-wismo-search-page-form"
  id="mimotrackerguestsearch"
  name="mimotrackerguestsearch"
  onSubmit={[Function]}
>
  <h1
    className="title"
  >
    Para verificar el estado de su pedido, ingrese su información a continuación.
  </h1>
  <label
    htmlFor="name"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Apellido o nombre comercial
    </div>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label
    htmlFor="order-number"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Número de orden
    </div>
    <input
      name="order-number"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onPaste={[Function]}
      value=""
    />
  </label>
  <div
    className="recaptcha"
  />
  <div
    className="dte-wismo-search-form-error"
  >
    <div
      className="message"
    >
      <span
        className="icon"
      >
        !
      </span>
      La información proporcionada no está asociada con un pedido activo. Inténtalo de nuevo.
    </div>
    <button
      className="close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="18"
        viewBox="0 0 20 20"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
          stroke="currentColor"
          strokeLinecap="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </div>
  <button
    className="button"
    data-track="mimo"
    data-track-action="click"
    data-track-detail="order tracker"
    data-track-sub-action="guest search"
    disabled={true}
    type="submit"
  >
    Enviar
  </button>
</form>
`;

exports[`<SearchPageForm /> should render correctly when error is ORDER_INVALID and language is en 1`] = `
<form
  className="dte-wismo-search-page-form"
  id="mimotrackerguestsearch"
  name="mimotrackerguestsearch"
  onSubmit={[Function]}
>
  <h1
    className="title"
  >
    To check the status of your order, please enter your information below.
  </h1>
  <label
    htmlFor="name"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Last Name or Business Name
    </div>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label
    htmlFor="order-number"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Order Number
    </div>
    <input
      name="order-number"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onPaste={[Function]}
      value=""
    />
  </label>
  <div
    className="recaptcha"
  />
  <div
    className="dte-wismo-search-form-error"
  >
    <div
      className="message"
    >
      <span
        className="icon"
      >
        !
      </span>
      The order number entered is not valid. Please try again.
    </div>
    <button
      className="close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="18"
        viewBox="0 0 20 20"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
          stroke="currentColor"
          strokeLinecap="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </div>
  <button
    className="button"
    data-track="mimo"
    data-track-action="click"
    data-track-detail="order tracker"
    data-track-sub-action="guest search"
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`<SearchPageForm /> should render correctly when error is ORDER_INVALID and language is es 1`] = `
<form
  className="dte-wismo-search-page-form"
  id="mimotrackerguestsearch"
  name="mimotrackerguestsearch"
  onSubmit={[Function]}
>
  <h1
    className="title"
  >
    Para verificar el estado de su pedido, ingrese su información a continuación.
  </h1>
  <label
    htmlFor="name"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Apellido o nombre comercial
    </div>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label
    htmlFor="order-number"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Número de orden
    </div>
    <input
      name="order-number"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onPaste={[Function]}
      value=""
    />
  </label>
  <div
    className="recaptcha"
  />
  <div
    className="dte-wismo-search-form-error"
  >
    <div
      className="message"
    >
      <span
        className="icon"
      >
        !
      </span>
      El número de pedido ingresado no es válido. Inténtalo de nuevo.
    </div>
    <button
      className="close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="18"
        viewBox="0 0 20 20"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
          stroke="currentColor"
          strokeLinecap="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </div>
  <button
    className="button"
    data-track="mimo"
    data-track-action="click"
    data-track-detail="order tracker"
    data-track-sub-action="guest search"
    disabled={true}
    type="submit"
  >
    Enviar
  </button>
</form>
`;

exports[`<SearchPageForm /> should render correctly when error is ORDER_NOT_FOUND and language is en 1`] = `
<form
  className="dte-wismo-search-page-form"
  id="mimotrackerguestsearch"
  name="mimotrackerguestsearch"
  onSubmit={[Function]}
>
  <h1
    className="title"
  >
    To check the status of your order, please enter your information below.
  </h1>
  <label
    htmlFor="name"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Last Name or Business Name
    </div>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label
    htmlFor="order-number"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Order Number
    </div>
    <input
      name="order-number"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onPaste={[Function]}
      value=""
    />
  </label>
  <div
    className="recaptcha"
  />
  <div
    className="dte-wismo-search-form-error"
  >
    <div
      className="message"
    >
      <span
        className="icon"
      >
        !
      </span>
      You have no active orders to track within our Order Tracker.
    </div>
    <button
      className="close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="18"
        viewBox="0 0 20 20"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
          stroke="currentColor"
          strokeLinecap="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </div>
  <button
    className="button"
    data-track="mimo"
    data-track-action="click"
    data-track-detail="order tracker"
    data-track-sub-action="guest search"
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`<SearchPageForm /> should render correctly when error is ORDER_NOT_FOUND and language is es 1`] = `
<form
  className="dte-wismo-search-page-form"
  id="mimotrackerguestsearch"
  name="mimotrackerguestsearch"
  onSubmit={[Function]}
>
  <h1
    className="title"
  >
    Para verificar el estado de su pedido, ingrese su información a continuación.
  </h1>
  <label
    htmlFor="name"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Apellido o nombre comercial
    </div>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label
    htmlFor="order-number"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Número de orden
    </div>
    <input
      name="order-number"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onPaste={[Function]}
      value=""
    />
  </label>
  <div
    className="recaptcha"
  />
  <div
    className="dte-wismo-search-form-error"
  >
    <div
      className="message"
    >
      <span
        className="icon"
      >
        !
      </span>
      No tiene órdenes activas para rastrear dentro de nuestro Rastreador de órdenes.
    </div>
    <button
      className="close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="18"
        viewBox="0 0 20 20"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
          stroke="currentColor"
          strokeLinecap="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </div>
  <button
    className="button"
    data-track="mimo"
    data-track-action="click"
    data-track-detail="order tracker"
    data-track-sub-action="guest search"
    disabled={true}
    type="submit"
  >
    Enviar
  </button>
</form>
`;

exports[`<SearchPageForm /> should render correctly when error is RECAPTCHA_FAILED and language is en 1`] = `
<form
  className="dte-wismo-search-page-form"
  id="mimotrackerguestsearch"
  name="mimotrackerguestsearch"
  onSubmit={[Function]}
>
  <h1
    className="title"
  >
    To check the status of your order, please enter your information below.
  </h1>
  <label
    htmlFor="name"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Last Name or Business Name
    </div>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label
    htmlFor="order-number"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Order Number
    </div>
    <input
      name="order-number"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onPaste={[Function]}
      value=""
    />
  </label>
  <div
    className="recaptcha"
  />
  <div
    className="dte-wismo-search-form-error"
  >
    <div
      className="message"
    >
      <span
        className="icon"
      >
        !
      </span>
      We apologize for the inconvenience, but reCAPTCHA is currently unavailable. Please try again later.
    </div>
    <button
      className="close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="18"
        viewBox="0 0 20 20"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
          stroke="currentColor"
          strokeLinecap="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </div>
  <button
    className="button"
    data-track="mimo"
    data-track-action="click"
    data-track-detail="order tracker"
    data-track-sub-action="guest search"
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`<SearchPageForm /> should render correctly when error is RECAPTCHA_FAILED and language is es 1`] = `
<form
  className="dte-wismo-search-page-form"
  id="mimotrackerguestsearch"
  name="mimotrackerguestsearch"
  onSubmit={[Function]}
>
  <h1
    className="title"
  >
    Para verificar el estado de su pedido, ingrese su información a continuación.
  </h1>
  <label
    htmlFor="name"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Apellido o nombre comercial
    </div>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label
    htmlFor="order-number"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Número de orden
    </div>
    <input
      name="order-number"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onPaste={[Function]}
      value=""
    />
  </label>
  <div
    className="recaptcha"
  />
  <div
    className="dte-wismo-search-form-error"
  >
    <div
      className="message"
    >
      <span
        className="icon"
      >
        !
      </span>
      Disculpe las molestias, pero reCAPTCHA no está disponible actualmente. Por favor, inténtelo de nuevo más tarde.
    </div>
    <button
      className="close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="18"
        viewBox="0 0 20 20"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
          stroke="currentColor"
          strokeLinecap="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </div>
  <button
    className="button"
    data-track="mimo"
    data-track-action="click"
    data-track-detail="order tracker"
    data-track-sub-action="guest search"
    disabled={true}
    type="submit"
  >
    Enviar
  </button>
</form>
`;

exports[`<SearchPageForm /> should render correctly when error is SERVER_UNAVAILABLE and language is en 1`] = `
<form
  className="dte-wismo-search-page-form"
  id="mimotrackerguestsearch"
  name="mimotrackerguestsearch"
  onSubmit={[Function]}
>
  <h1
    className="title"
  >
    To check the status of your order, please enter your information below.
  </h1>
  <label
    htmlFor="name"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Last Name or Business Name
    </div>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label
    htmlFor="order-number"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Order Number
    </div>
    <input
      name="order-number"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onPaste={[Function]}
      value=""
    />
  </label>
  <div
    className="recaptcha"
  />
  <div
    className="dte-wismo-search-form-error"
  >
    <div
      className="message"
    >
      <span
        className="icon"
      >
        !
      </span>
      Our system is currently unavailable. We apologize for the inconvenience. Please try again later.
    </div>
    <button
      className="close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="18"
        viewBox="0 0 20 20"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
          stroke="currentColor"
          strokeLinecap="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </div>
  <button
    className="button"
    data-track="mimo"
    data-track-action="click"
    data-track-detail="order tracker"
    data-track-sub-action="guest search"
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`<SearchPageForm /> should render correctly when error is SERVER_UNAVAILABLE and language is es 1`] = `
<form
  className="dte-wismo-search-page-form"
  id="mimotrackerguestsearch"
  name="mimotrackerguestsearch"
  onSubmit={[Function]}
>
  <h1
    className="title"
  >
    Para verificar el estado de su pedido, ingrese su información a continuación.
  </h1>
  <label
    htmlFor="name"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Apellido o nombre comercial
    </div>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label
    htmlFor="order-number"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Número de orden
    </div>
    <input
      name="order-number"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onPaste={[Function]}
      value=""
    />
  </label>
  <div
    className="recaptcha"
  />
  <div
    className="dte-wismo-search-form-error"
  >
    <div
      className="message"
    >
      <span
        className="icon"
      >
        !
      </span>
      Nuestro sistema no está disponible actualmente. Disculpe las molestias. Por favor, inténtelo de nuevo más tarde.
    </div>
    <button
      className="close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="18"
        viewBox="0 0 20 20"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
          stroke="currentColor"
          strokeLinecap="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </div>
  <button
    className="button"
    data-track="mimo"
    data-track-action="click"
    data-track-detail="order tracker"
    data-track-sub-action="guest search"
    disabled={true}
    type="submit"
  >
    Enviar
  </button>
</form>
`;

exports[`<SearchPageForm /> should render correctly when error is well this was unexpected and language is en 1`] = `
<form
  className="dte-wismo-search-page-form"
  id="mimotrackerguestsearch"
  name="mimotrackerguestsearch"
  onSubmit={[Function]}
>
  <h1
    className="title"
  >
    To check the status of your order, please enter your information below.
  </h1>
  <label
    htmlFor="name"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Last Name or Business Name
    </div>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label
    htmlFor="order-number"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Order Number
    </div>
    <input
      name="order-number"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onPaste={[Function]}
      value=""
    />
  </label>
  <div
    className="recaptcha"
  />
  <div
    className="dte-wismo-search-form-error"
  >
    <div
      className="message"
    >
      <span
        className="icon"
      >
        !
      </span>
      An unexpected error has occured. We apologize for the inconvenience. Please try again later.
    </div>
    <button
      className="close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="18"
        viewBox="0 0 20 20"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
          stroke="currentColor"
          strokeLinecap="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </div>
  <button
    className="button"
    data-track="mimo"
    data-track-action="click"
    data-track-detail="order tracker"
    data-track-sub-action="guest search"
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`<SearchPageForm /> should render correctly when error is well this was unexpected and language is es 1`] = `
<form
  className="dte-wismo-search-page-form"
  id="mimotrackerguestsearch"
  name="mimotrackerguestsearch"
  onSubmit={[Function]}
>
  <h1
    className="title"
  >
    Para verificar el estado de su pedido, ingrese su información a continuación.
  </h1>
  <label
    htmlFor="name"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Apellido o nombre comercial
    </div>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label
    htmlFor="order-number"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Número de orden
    </div>
    <input
      name="order-number"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onPaste={[Function]}
      value=""
    />
  </label>
  <div
    className="recaptcha"
  />
  <div
    className="dte-wismo-search-form-error"
  >
    <div
      className="message"
    >
      <span
        className="icon"
      >
        !
      </span>
      Ha ocurrido un error inesperado. Disculpe las molestias. Por favor, inténtelo de nuevo más tarde.
    </div>
    <button
      className="close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="18"
        viewBox="0 0 20 20"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
          stroke="currentColor"
          strokeLinecap="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </div>
  <button
    className="button"
    data-track="mimo"
    data-track-action="click"
    data-track-detail="order tracker"
    data-track-sub-action="guest search"
    disabled={true}
    type="submit"
  >
    Enviar
  </button>
</form>
`;

exports[`<SearchPageForm /> should render correctly when language is en 1`] = `
<form
  className="dte-wismo-search-page-form"
  id="mimotrackerguestsearch"
  name="mimotrackerguestsearch"
  onSubmit={[Function]}
>
  <h1
    className="title"
  >
    To check the status of your order, please enter your information below.
  </h1>
  <label
    htmlFor="name"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Last Name or Business Name
    </div>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label
    htmlFor="order-number"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Order Number
    </div>
    <input
      name="order-number"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onPaste={[Function]}
      value=""
    />
  </label>
  <div
    className="recaptcha"
  />
  <button
    className="button"
    data-track="mimo"
    data-track-action="click"
    data-track-detail="order tracker"
    data-track-sub-action="guest search"
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`<SearchPageForm /> should render correctly when language is es 1`] = `
<form
  className="dte-wismo-search-page-form"
  id="mimotrackerguestsearch"
  name="mimotrackerguestsearch"
  onSubmit={[Function]}
>
  <h1
    className="title"
  >
    Para verificar el estado de su pedido, ingrese su información a continuación.
  </h1>
  <label
    htmlFor="name"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Apellido o nombre comercial
    </div>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label
    htmlFor="order-number"
  >
    <div
      className="dte-wismo-search-page-form-input-header"
    >
      Número de orden
    </div>
    <input
      name="order-number"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onPaste={[Function]}
      value=""
    />
  </label>
  <div
    className="recaptcha"
  />
  <button
    className="button"
    data-track="mimo"
    data-track-action="click"
    data-track-detail="order tracker"
    data-track-sub-action="guest search"
    disabled={true}
    type="submit"
  >
    Enviar
  </button>
</form>
`;
